spec/unit/data_providers/hiera_data_provider_spec.rb

```diff
@@ -190,6 +190,90 @@ def compile(environment, code = nil)
       end
     end
   end
+
+  describe 'when using explain' do
+    it 'will report config path (original and resolved), data path (original and resolved), and interpolation (before and after)' do
+      compile('hiera_misc', '$target_scope = "with scope"') do |compiler|
+        lookup_invocation = Puppet::Pops::Lookup::Invocation.new(compiler.topscope, {}, {}, true)
+        value = Puppet::Pops::Lookup.lookup('one::lopts_test::hash', nil, nil, nil, nil, lookup_invocation)
+        expect(lookup_invocation.explainer.to_s).to eq(<<EOS)
+Searching for 'lookup_options'
+  Merge strategy hash
+    Data Binding "hiera"
+      No such key: "lookup_options"
+    Data Provider "Hiera Data Provider, version 4"
+      ConfigurationPath "#{environmentpath}/hiera_misc/hiera.yaml"
+      Data Provider "common"
+        Path "#{environmentpath}/hiera_misc/data/common.yaml"
+          Original path: common
+          Found key: "lookup_options" value: {
+            "one::lopts_test::hash" => {
```

Comment:
If we do, then we loose the ability to view the complete set of lookup options for a specific module. We could have a specific option for that of course, but the way it stands now, the --explain will give a full explanation. Subsequent lookups may well use --explain-data which doesn't include lookup_options at all.

Another way of doing this would be to all of the following:

1. Not allow lookup of 'lookup_options' (it's not 'data', it's 'meta-data').
2. Provide a specific option that would let you view all 'lookup_options' for a given module.
3. Never display 'lookup_options' when using --explain.

Pros: Less clutter, more specific, less confusing.
Cons: You see 'Merge strategy xxx' when using --explain but have to consult the 'lookup_options' merge result from another comman to see where it actually came from.